offscreen-canvas/tools/gentest.py

```diff
@@ -0,0 +1,757 @@
+# This script is modified based on 2dcontext/tools/gentest.py
```

Comment:
Could we avoid so much code duplication? I think you should factor-out all the shared stuff into a module that is imported by both generators.  That will make these tests much easier to maintain in the future, and it will prevent them from forking apart.